### PR TITLE
Feature/kitosudv 2219 task 2318 procurement

### PR DESCRIPTION
**Pull request by JacobMalling**
## KITOS Pull request template
The following procedure dictates the steps needed before a Pull request can be merged into master.

- [x] **Implement**: 
      *All requirements are implemented and unit tests are green*
      
- [x] **Merge master into branch / rebase with master**: 
      *Make sure you are testing your changes and how they co-exist with the latest version of master*
      
- [x] **Green on integration**: 
      *All integration tests are green on integration*
      
- [x] **Request review**: 
      *Tag whomever you wish to review your code*
      
- [x] **Review completed**: 
      *Reviewer ticks this box when review comments have been submitted*
      
- [x] **Changes**: 
      *PR owner and reviewer agrees on which changes must be made and the changes are committed.*
      
- [x] **Merge master into branch / rebase with master**: 
      *Make sure you are testing your changes and how they co-exist with the latest version of master*
      
- [x] **Green on integration**: 
      *All integration tests are green on integration*

### CODE REVIEW

File: Presentation.Web/Controllers/API/V2/External/ItContracts/Mapping/ItContractWriteModelMapper.cs [hunk 1]
```diff
@@ -54,5 +56,16 @@ public ItContractGeneralDataModificationParameters MapGeneralData(ContractGenera
...
             };
         }
+
+        public ItContractProcurementModificationParameters MapProcurement(ContractProcurementDataWriteRequestDTO request)
+        {
+            return new()
+            {
+                ProcurementStrategyUuid = request.ProcurementStrategyUuid.AsChangedValue(),
+                PurchaseTypeUuid = request.PurchaseTypeUuid.AsChangedValue(),
+                HalfOfYear = (request.ProcurementPlan?.HalfOfYear ?? Maybe<byte>.None).AsChangedValue(),
```
[the remark this answers is not in the document]
Comment (mrjsawdk) [reply Presentation.Web/Controllers/API/V2/External/ItContracts/Mapping/ItContractWriteModelMapper.cs:66]: Så vil alle de tidligere nævnte problemer også forsvinde

**Review by mrjsawdk — APPROVED** [review]
lidt justeringer og ellers cool

File: Core.ApplicationServices/Contract/Write/ItContractWriteService.cs [hunk 2]
```diff
@@ -273,5 +288,37 @@ public Maybe<OperationError> Delete(Guid itContractUuid)
                 .Delete(dbId.Value)
                 .Match(_ => Maybe<OperationError>.None, failure => new OperationError("Failed deleting contract", failure));
         }
+
+        private Maybe<OperationError> UpdateIndependentOptionTypeAssignment<TOption>(
```
[the remark this answers is not in the document]
Comment (JacobMalling) [reply Core.ApplicationServices/Contract/Write/ItContractWriteService.cs:292]: Jeg havde min egen udgave af den. Der skulle kun være en enkelt af dem tilbage i test klassen

[the remark this answers is not in the document]
Comment (mrjsawdk) [reply Core.ApplicationServices/Contract/Write/ItContractWriteService.cs:292]: den står bare som en tilføjelse så det er lidt pudsigt

[the remark this answers is not in the document]
Comment (mrjsawdk) [reply Core.ApplicationServices/Contract/Write/ItContractWriteService.cs:292]: ahh jeg kan se at den er flyttet

**Pull request merged**